pkgs/by-name/mc/mcumgr-client/package.nix

```diff
@@ -0,0 +1,44 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  nix-update-script,
+  pkg-config,
+  udev,
+  stdenv,
+  darwin,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "mcumgr-client";
+  version = "0.0.4";
+
+  src = fetchFromGitHub {
+    owner = "vouch-opensource";
+    repo = "mcumgr-client";
+    rev = "v${version}";
+    hash = "sha256-MTNMnA5/CzwVrhNhDrfaXOatT4BFmc4nOPhIxTyc248=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
+
+  passthru.updateScript = nix-update-script { };
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ udev ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.IOKit ];
```

Comment:
`udev` is actually systemd, it's also not available on darwin.

Reply:
I made it Linux-specific.